gix-worktree/src/index/status/diff.rs

```diff
@@ -0,0 +1,87 @@
+use gix_features::hash;
+use gix_hash::ObjectId;
+use gix_index as index;
+use gix_object::encode::loose_header;
+use index::Entry;
+
+///
+pub trait LazyBlob<'a, E> {
+    ///
+    fn read(self) -> Result<&'a [u8], E>;
+}
+
+///
+pub trait Diff: Send + Sync {
```

Comment:
Empty comments are very dangerous. Rather add `#[allow(missing_docs)]` where needed.

These docs would now be particularly called for as I don't think `Diff` is a very descriptive name. Using the convention of single-method-traits it could be called `ContentChanged` or maybe even `HasContentChanged::has_content_changed()`.

Reply:
I renamed this and added some documentation